src/rgw/rgw_sal.cc

```diff
@@ -98,6 +98,9 @@ rgw::sal::Store* StoreManager::init_storage_provider(const DoutPrefixProvider* d
                 .initialize(cct, dpp) < 0) {
       delete store; store = nullptr;
     }
+    if (store->initialize(cct, dpp) < 0) {
+      delete store; store = nullptr;
```

Comment:
And this.

Reply:
Okay, it turns out this is hard, because of RGWRealmReloader, that keeps a reference to main()'s store pointer (!) and replaces it sometimes.  This is, I think, going go be a big problem for stacking and so on, but also means main can't have a unique_ptr without extensive changes.  I'll put unique_ptr of store on my list for later.

Reply:
sounds good. i'm happy to discuss the requirements for `RGWRealmReloader` when we get there